snapcraft/_store.py

```diff
@@ -59,19 +112,24 @@ def _login(store, acls=None, save=True):
         try:
             store.login(email, password, acls=acls, save=save)
             print()
-            logger.info(
-                'We strongly recommend enabling multi-factor authentication: '
-                'https://help.ubuntu.com/community/SSO/FAQs/2FA')
+            logger.info(storeapi.constants.TWO_FACTOR_WARNING)
         except storeapi.errors.StoreTwoFactorAuthenticationRequired:
             one_time_password = input('Second-factor auth: ')
             store.login(
                 email, password, one_time_password=one_time_password,
                 acls=acls, save=save)
-    except (storeapi.errors.InvalidCredentialsError,
-            storeapi.errors.StoreAuthenticationError):
-        print()
-        logger.info('Login failed.')
-        return False
+
+        # Continue if agreement and namespace conditions are met.
+        _check_dev_agreement_and_namespace_statuses(store)
+
+    except storeapi.errors.InvalidCredentialsError:
+        return _fail_login(storeapi.constants.INVALID_CREDENTIALS)
+    except storeapi.errors.StoreAuthenticationError:
+        return _fail_login(storeapi.constants.AUTHENTICATION_ERROR)
+    except storeapi.errors.StoreAccountInformationError:
+        return _fail_login(storeapi.constants.ACCOUNT_INFORMATION_ERROR)
+    except storeapi.errors.NeedTermsSignedError as e:
+        return _fail_login(e.message)
```

Comment:
The `NeedTermsSignedError` will contain different messages based on the error in `_check_dev_agreement_and_namespace_statuses`.